genie/input_to_database.py

```diff
@@ -464,6 +465,69 @@ def create_and_archive_maf_database(syn, database_synid_mappingdf):
     return(database_synid_mappingdf)
 
 
+def email_duplication_error(syn, duplicated_filesdf):
+    '''
+    Sends an email if there is a duplication error
+
+    Args:
+        syn: Synapse object
+        duplicated_filesdf: dataframe with 'id', 'name' column
+    '''
+    if not duplicated_filesdf.empty:
+        incorrect_files = [
+            name for synId, name in zip(duplicated_filesdf['id'],
+                                        duplicated_filesdf['name'])]
+        incorrect_filenames = ", ".join(incorrect_files)
+        incorrect_ent = syn.get(duplicated_filesdf['id'].iloc[0])
+        send_to_users = set([incorrect_ent.modifiedBy,
+                             incorrect_ent.createdBy])
+        usernames = ", ".join(
+            [syn.getUserProfile(user)['userName'] for user in send_to_users])
+        error_email = (
+            "Dear %s,\n\n"
+            "Your files (%s) are duplicated!  FILES SHOULD BE UPLOADED AS "
+            "NEW VERSIONS AND THE ENTIRE DATASET SHOULD BE "
+            "UPLOADED EVERYTIME".format(usernames, incorrect_filenames))
+        syn.sendMessage(
+            list(send_to_users), "GENIE Validation Error", error_email)
+
+
+def get_duplicated_files(syn, validation_statusdf, duplicated_error_message):
```

Comment:
can you do this way earlier, prior to file validation?

Reply:
Harder to move out than I thought...  The issue is the duplication errors also get saved the invalid status table which has information that is extracted during the validation call (md5, modifiedOn).

This reordering would be possible by building up the information of the entity upon my first rest call. #135.